src/udev/udevd.c

```diff
@@ -1798,7 +1798,7 @@ static int run(int argc, char *argv[]) {
 
         dev_setup(NULL, UID_INVALID, GID_INVALID);
 
-        if (getppid() == 1) {
+        if (getppid() == 1 && sd_booted() > 0) {
```

Comment:
`sd_booted` is documented to be able to return negative errno on failure. It currently doesn't, but I think if sd_booted should ever return an error, udev should not enable the killing logic. Better to leave stray processes than to kill unrelated stuff.

Reply:
`sd_booted()` does not return negative errno. I've prepared PR #11661 for updating the document.

Reply:
hmm, i think we should fix sd_booted() instead to return an error if we can't figure out whether /run/systemd/system/ exists

Reply:
@poettering #11665.